app/components/post_list/post_list.test.tsx

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {act, fireEvent} from '@testing-library/react-native';
+import React from 'react';
+import {DeviceEventEmitter, Platform} from 'react-native';
+
+import * as localPostFunctions from '@actions/local/post';
+import * as postFunctions from '@actions/remote/post';
+import {Events, Screens} from '@constants';
+import {renderWithEverything} from '@test/intl-test-helper';
+import TestHelper from '@test/test_helper';
+
+import PostList, {exportedForTesting} from './post_list';
+
+jest.mock('@components/post_list/post', () => 'Post');
+jest.mock('@components/post_list/thread_overview', () => 'ThreadOverview');
+jest.mock('@components/post_list/more_messages', () => 'MoreMessages');
+jest.mock('@components/post_list/combined_user_activity', () => 'CombinedUserActivity');
+jest.mock('@components/post_list/scroll_to_end_view', () => 'ScrollToEndView');
+jest.mock('@actions/remote/post', () => {
+    return {
+        fetchPosts: jest.fn(),
+        fetchPostThread: jest.fn(),
+    };
+});
+jest.mock('@actions/local/post', () => ({
+    removePost: jest.fn(),
+}));
+
+import type {PostModel} from '@database/models/server';
+import type Database from '@nozbe/watermelondb/Database';
+
+const {PostListProps} = exportedForTesting;
+
+describe('components/post_list/PostList', () => {
+    let database: Database;
+    const serverUrl = 'https://server.com';
+    const fetchPostsSpy = jest.spyOn(postFunctions, 'fetchPosts');
+    const fetchPostThreadSpy = jest.spyOn(postFunctions, 'fetchPostThread');
+    const removePostSpy = jest.spyOn(localPostFunctions, 'removePost');
+
+    beforeAll(async () => {
+        const server = await TestHelper.setupServerDatabase();
+        database = server.database;
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        Platform.select = jest.fn().mockImplementation((obj) => obj.ios);
+    });
+
+    const mockPostModel = (overrides: Partial<PostModel> = {}): PostModel => ({
+        id: 'post-id',
+        channelId: 'channel-id',
+        createAt: Date.now(),
+        deleteAt: 0,
+        type: 'custom_post_type',
+        userId: 'user-id',
+        ...overrides,
+    } as PostModel);
```

Comment:
I copied this from one of @enahum's tests.  I should consider making a helper utility function and share instead of making another copy.